framework/src/base/MooseApp.C

```diff
@@ -1067,19 +1083,8 @@ MooseApp::getCheckpointDirectories() const
       continue;
 
     const InputParameters & params = moose_object_action->getObjectParams();
-
-    // Loop through the actions and add the necessary directories to the list to check
-    if (moose_object_action->getParamTempl<std::string>("type") == "Checkpoint")
```

Comment:
Interesting, how did we manage to get rid of this condition? Are we getting extra garbage in the `checkpoint_dirs` vector that is just getting ignored?